// ionic2-calendar/calendar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../node_modules/ionic-angular/components/item/item.ngfactory';
import * as import2 from 'ionic-angular/components/item/item';
import * as import3 from 'ionic-angular/util/form';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/components/item/item-reorder';
import * as import6 from 'ionic-angular/components/item/item-content';
import * as import7 from '@angular/common';
import * as import8 from 'ionic-angular/components/list/list';
import * as import9 from 'ionic-angular/platform/platform';
import * as import10 from 'ionic-angular/gestures/gesture-controller';
import * as import11 from 'ionic-angular/platform/dom-controller';
import * as import12 from './monthview.ngfactory';
import * as import13 from './monthview';
import * as import14 from './calendar.service';
import * as import15 from './weekview.ngfactory';
import * as import16 from './weekview';
import * as import17 from './dayview.ngfactory';
import * as import18 from './dayview';
import * as import19 from './calendar';
const styles_CalendarComponent:any[] = ['[_nghost-%COMP%]    > div[_ngcontent-%COMP%] { height: 100%; }\n\n        .event-detail-container[_ngcontent-%COMP%] {\n          border-top: 2px darkgrey solid;\n        }\n\n        .no-events-label[_ngcontent-%COMP%] {\n          font-weight: bold;\n          color: darkgrey;\n          text-align: center;\n        }\n\n        .event-detail[_ngcontent-%COMP%] {\n          cursor: pointer;\n          white-space: nowrap;\n          text-overflow: ellipsis;\n        }\n\n        .monthview-eventdetail-timecolumn[_ngcontent-%COMP%] {\n          width: 110px;\n          overflow: hidden;\n        }\n\n        .calendar-event-inner[_ngcontent-%COMP%] {\n          overflow: hidden;\n          background-color: #3a87ad;\n          color: white;\n          height: 100%;\n          width: 100%;\n          padding: 2px;\n          line-height: 15px;\n        }\n\n        @media (max-width: 750px) {\n          .calendar-event-inner[_ngcontent-%COMP%] {\n            font-size: 12px;\n          }\n        }'];
export const RenderType_CalendarComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_CalendarComponent,
  data: {}
}
);
function View_CalendarComponent_1(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '\n            ',
      '\n        '
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = v.context.view.dates[((v.context.row * 7) + v.context.col)].label;
    ck(v,0,0,currVal_0);
  });
}
function View_CalendarComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'monthview-eventdetail-timecolumn'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '\n                            -\n                            ',
      '\n                        '
    ]
    )),
    import0.ɵppd(2),
    import0.ɵppd(2)
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵunv(v,1,0,ck(v,2,0,import0.ɵnov((<any>(<any>(<any>(<any>v.parent).parent).parent).parent),0),(<any>v.parent).context.$implicit.startTime,'HH:mm'));
    const currVal_1:any = import0.ɵunv(v,1,1,ck(v,3,0,import0.ɵnov((<any>(<any>(<any>(<any>v.parent).parent).parent).parent),0),(<any>v.parent).context.$implicit.endTime,'HH:mm'));
    ck(v,1,0,currVal_0,currVal_1);
  });
}
function View_CalendarComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'monthview-eventdetail-timecolumn'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.allDayLabel;
    ck(v,1,0,currVal_0);
  });
}
function View_CalendarComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),15,'ion-item',[[
        'class',
        'item item-block'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import1.View_Item_0,import1.RenderType_Item)),
    import0.ɵdid(548864,(null as any),3,import2.Item,[
      import3.Form,
      import4.Config,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import5.ItemReorder
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{contentLabel: 0}),
    import0.ɵqud(301989888,2,{_buttons: 1}),
    import0.ɵqud(301989888,3,{_icons: 1}),
    import0.ɵdid(8192,(null as any),0,import6.ItemContent,([] as any[]),(null as any),(null as any)),
    (l()(),import0.ɵted(2,['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),2,1,(null as any),View_CalendarComponent_5)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(2,['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),2,1,(null as any),View_CalendarComponent_6)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(2,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),2,1,'span',[[
        'class',
        'event-detail'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '  |  ',
      ''
    ]
    )),
    (l()(),import0.ɵted(2,['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:boolean = !v.context.$implicit.allDay;
    ck(v,8,0,currVal_0);
    const currVal_1:any = v.context.$implicit.allDay;
    ck(v,11,0,currVal_1);
  },(ck,v) => {
    const currVal_2:any = v.context.$implicit.title;
    ck(v,14,0,currVal_2);
  });
}
function View_CalendarComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'ion-item',[[
        'class',
        'item item-block'
      ]
    ],(null as any),(null as any),(null as any),import1.View_Item_0,import1.RenderType_Item)),
    import0.ɵdid(548864,(null as any),3,import2.Item,[
      import3.Form,
      import4.Config,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import5.ItemReorder
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,4,{contentLabel: 0}),
    import0.ɵqud(301989888,5,{_buttons: 1}),
    import0.ɵqud(301989888,6,{_icons: 1}),
    import0.ɵdid(8192,(null as any),0,import6.ItemContent,([] as any[]),(null as any),(null as any)),
    (l()(),import0.ɵted(2,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),2,1,'div',[[
        'class',
        'no-events-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted(2,['\n                ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.noEventsLabel;
    ck(v,8,0,currVal_0);
  });
}
function View_CalendarComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'ion-list',[
      [
        'class',
        'event-detail-container'
      ]
      ,
      [
        'has-bouncing',
        'false'
      ]
      ,
      [
        'overflow-scroll',
        'false'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import8.List,[
      import4.Config,
      import0.ElementRef,
      import0.Renderer,
      import9.Platform,
      import10.GestureController,
      import11.DomController
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_4)),
    import0.ɵdid(401408,(null as any),0,import7.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_7)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = (((<any>v.parent).context.selectedDate == null)? (null as any): (<any>v.parent).context.selectedDate.events);
    ck(v,4,0,currVal_0);
    const currVal_1:any = ((((<any>v.parent).context.selectedDate == null)? (null as any): (<any>v.parent).context.selectedDate.events.length) == 0);
    ck(v,7,0,currVal_1);
  },(null as any));
}
function View_CalendarComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_3)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.showEventDetail;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_CalendarComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'calendar-event-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.displayEvent.event.title;
    ck(v,2,0,currVal_0);
  });
}
function View_CalendarComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'calendar-event-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.displayEvent.event.title;
    ck(v,2,0,currVal_0);
  });
}
function View_CalendarComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_CalendarComponent_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import7.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_CalendarComponent_13)),
    import0.ɵdid(270336,(null as any),0,import7.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,((25 * v.context.$implicit.position) + 'px'),((100 * (v.context.$implicit.endIndex - v.context.$implicit.startIndex)) + '%'),'25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>(<any>v.parent).parent).context.eventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_CalendarComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import7.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import7.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_12)),
    import0.ɵdid(401408,(null as any),0,import7.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.day.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = ck(v,4,0,((25 * (<any>v.parent).context.day.events.length) + 'px'));
    ck(v,3,0,currVal_1);
    const currVal_2:any = (<any>v.parent).context.day.events;
    ck(v,7,0,currVal_2);
  },(null as any));
}
function View_CalendarComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_11)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.day.events;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_CalendarComponent_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_CalendarComponent_15(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import7.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_CalendarComponent_16)),
    import0.ɵdid(270336,(null as any),0,import7.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,((25 * v.context.index) + 'px'),'100%','25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.eventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_CalendarComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_15)),
    import0.ɵdid(401408,(null as any),0,import7.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.allDayEvents;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_CalendarComponent_20(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_CalendarComponent_19(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import7.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_CalendarComponent_20)),
    import0.ɵdid(270336,(null as any),0,import7.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / (<any>(<any>v.parent).parent).context.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / (<any>(<any>v.parent).parent).context.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>(<any>v.parent).parent).context.eventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_CalendarComponent_18(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import7.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_19)),
    import0.ɵdid(401408,(null as any),0,import7.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.tm.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.tm.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_CalendarComponent_17(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_18)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.tm.events;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_CalendarComponent_21(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'monthview',([] as any[]),(null as any),[
      [
        (null as any),
        'onRangeChanged'
      ]
      ,
      [
        (null as any),
        'onEventSelected'
      ]
      ,
      [
        (null as any),
        'onTimeSelected'
      ]
      ,
      [
        (null as any),
        'onTitleChanged'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onRangeChanged' === en)) {
        const pd_0:any = ((<any>co.rangeChanged($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('onEventSelected' === en)) {
        const pd_1:any = ((<any>co.eventSelected($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('onTimeSelected' === en)) {
        const pd_2:any = ((<any>co.timeSelected($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('onTitleChanged' === en)) {
        const pd_3:any = ((<any>co.titleChanged($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },import12.View_MonthViewComponent_0,import12.RenderType_MonthViewComponent)),
    import0.ɵdid(2482176,(null as any),0,import13.MonthViewComponent,[import14.CalendarService],{
      monthviewDisplayEventTemplate: [
        0,
        'monthviewDisplayEventTemplate'
      ]
      ,
      monthviewInactiveDisplayEventTemplate: [
        1,
        'monthviewInactiveDisplayEventTemplate'
      ]
      ,
      monthviewEventDetailTemplate: [
        2,
        'monthviewEventDetailTemplate'
      ]
      ,
      formatDay: [
        3,
        'formatDay'
      ]
      ,
      formatDayHeader: [
        4,
        'formatDayHeader'
      ]
      ,
      formatMonthTitle: [
        5,
        'formatMonthTitle'
      ]
      ,
      eventSource: [
        6,
        'eventSource'
      ]
      ,
      startingDayMonth: [
        7,
        'startingDayMonth'
      ]
      ,
      showEventDetail: [
        8,
        'showEventDetail'
      ]
      ,
      noEventsLabel: [
        9,
        'noEventsLabel'
      ]
      ,
      autoSelect: [
        10,
        'autoSelect'
      ]
      ,
      markDisabled: [
        11,
        'markDisabled'
      ]
      ,
      locale: [
        12,
        'locale'
      ]
      ,
      dateFormatter: [
        13,
        'dateFormatter'
      ]
      ,
      dir: [
        14,
        'dir'
      ]
      ,
      lockSwipeToPrev: [
        15,
        'lockSwipeToPrev'
      ]
      ,
      lockSwipes: [
        16,
        'lockSwipes'
      ]
      ,
      spaceBetween: [
        17,
        'spaceBetween'
      ]

    }
    ,{
      onRangeChanged: 'onRangeChanged',
      onEventSelected: 'onEventSelected',
      onTimeSelected: 'onTimeSelected',
      onTitleChanged: 'onTitleChanged'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.monthviewDisplayEventTemplate || import0.ɵnov((<any>v.parent),2));
    const currVal_1:any = (co.monthviewInactiveDisplayEventTemplate || import0.ɵnov((<any>v.parent),2));
    const currVal_2:any = (co.monthviewEventDetailTemplate || import0.ɵnov((<any>v.parent),4));
    const currVal_3:any = co.formatDay;
    const currVal_4:any = co.formatDayHeader;
    const currVal_5:any = co.formatMonthTitle;
    const currVal_6:any = co.eventSource;
    const currVal_7:any = co.startingDayMonth;
    const currVal_8:any = co.showEventDetail;
    const currVal_9:any = co.noEventsLabel;
    const currVal_10:any = co.autoSelect;
    const currVal_11:any = co.markDisabled;
    const currVal_12:any = co.locale;
    const currVal_13:any = co.dateFormatter;
    const currVal_14:any = co.dir;
    const currVal_15:any = co.lockSwipeToPrev;
    const currVal_16:any = co.lockSwipes;
    const currVal_17:any = co.spaceBetween;
    ck(v,1,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17
    ]
    );
  },(null as any));
}
function View_CalendarComponent_22(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'weekview',([] as any[]),[[
        2,
        'weekview',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'onRangeChanged'
      ]
      ,
      [
        (null as any),
        'onEventSelected'
      ]
      ,
      [
        (null as any),
        'onTimeSelected'
      ]
      ,
      [
        (null as any),
        'onTitleChanged'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onRangeChanged' === en)) {
        const pd_0:any = ((<any>co.rangeChanged($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('onEventSelected' === en)) {
        const pd_1:any = ((<any>co.eventSelected($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('onTimeSelected' === en)) {
        const pd_2:any = ((<any>co.timeSelected($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('onTitleChanged' === en)) {
        const pd_3:any = ((<any>co.titleChanged($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },import15.View_WeekViewComponent_0,import15.RenderType_WeekViewComponent)),
    import0.ɵdid(2482176,(null as any),0,import16.WeekViewComponent,[
      import14.CalendarService,
      import0.ElementRef
    ]
    ,{
      weekviewAllDayEventTemplate: [
        0,
        'weekviewAllDayEventTemplate'
      ]
      ,
      weekviewNormalEventTemplate: [
        1,
        'weekviewNormalEventTemplate'
      ]
      ,
      weekviewAllDayEventSectionTemplate: [
        2,
        'weekviewAllDayEventSectionTemplate'
      ]
      ,
      weekviewNormalEventSectionTemplate: [
        3,
        'weekviewNormalEventSectionTemplate'
      ]
      ,
      formatWeekTitle: [
        4,
        'formatWeekTitle'
      ]
      ,
      formatWeekViewDayHeader: [
        5,
        'formatWeekViewDayHeader'
      ]
      ,
      formatHourColumn: [
        6,
        'formatHourColumn'
      ]
      ,
      startingDayWeek: [
        7,
        'startingDayWeek'
      ]
      ,
      allDayLabel: [
        8,
        'allDayLabel'
      ]
      ,
      hourParts: [
        9,
        'hourParts'
      ]
      ,
      eventSource: [
        10,
        'eventSource'
      ]
      ,
      markDisabled: [
        11,
        'markDisabled'
      ]
      ,
      locale: [
        12,
        'locale'
      ]
      ,
      dateFormatter: [
        13,
        'dateFormatter'
      ]
      ,
      dir: [
        14,
        'dir'
      ]
      ,
      scrollToHour: [
        15,
        'scrollToHour'
      ]
      ,
      preserveScrollPosition: [
        16,
        'preserveScrollPosition'
      ]
      ,
      lockSwipeToPrev: [
        17,
        'lockSwipeToPrev'
      ]
      ,
      lockSwipes: [
        18,
        'lockSwipes'
      ]
      ,
      startHour: [
        19,
        'startHour'
      ]
      ,
      endHour: [
        20,
        'endHour'
      ]
      ,
      spaceBetween: [
        21,
        'spaceBetween'
      ]
      ,
      hourSegments: [
        22,
        'hourSegments'
      ]

    }
    ,{
      onRangeChanged: 'onRangeChanged',
      onEventSelected: 'onEventSelected',
      onTimeSelected: 'onTimeSelected',
      onTitleChanged: 'onTitleChanged'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.weekviewAllDayEventTemplate || import0.ɵnov((<any>v.parent),6));
    const currVal_2:any = (co.weekviewNormalEventTemplate || import0.ɵnov((<any>v.parent),8));
    const currVal_3:any = (co.weekviewAllDayEventSectionTemplate || import0.ɵnov((<any>v.parent),10));
    const currVal_4:any = (co.weekviewNormalEventSectionTemplate || import0.ɵnov((<any>v.parent),14));
    const currVal_5:any = co.formatWeekTitle;
    const currVal_6:any = co.formatWeekViewDayHeader;
    const currVal_7:any = co.formatHourColumn;
    const currVal_8:any = co.startingDayWeek;
    const currVal_9:any = co.allDayLabel;
    const currVal_10:any = co.hourParts;
    const currVal_11:any = co.eventSource;
    const currVal_12:any = co.markDisabled;
    const currVal_13:any = co.locale;
    const currVal_14:any = co.dateFormatter;
    const currVal_15:any = co.dir;
    const currVal_16:any = co.scrollToHour;
    const currVal_17:any = co.preserveScrollPosition;
    const currVal_18:any = co.lockSwipeToPrev;
    const currVal_19:any = co.lockSwipes;
    const currVal_20:any = co.startHour;
    const currVal_21:any = co.endHour;
    const currVal_22:any = co.spaceBetween;
    const currVal_23:any = co.hourSegments;
    ck(v,1,1,[
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19,
      currVal_20,
      currVal_21,
      currVal_22,
      currVal_23
    ]
    );
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).class;
    ck(v,0,0,currVal_0);
  });
}
function View_CalendarComponent_23(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'dayview',([] as any[]),[[
        2,
        'dayview',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'onRangeChanged'
      ]
      ,
      [
        (null as any),
        'onEventSelected'
      ]
      ,
      [
        (null as any),
        'onTimeSelected'
      ]
      ,
      [
        (null as any),
        'onTitleChanged'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onRangeChanged' === en)) {
        const pd_0:any = ((<any>co.rangeChanged($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('onEventSelected' === en)) {
        const pd_1:any = ((<any>co.eventSelected($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('onTimeSelected' === en)) {
        const pd_2:any = ((<any>co.timeSelected($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('onTitleChanged' === en)) {
        const pd_3:any = ((<any>co.titleChanged($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },import17.View_DayViewComponent_0,import17.RenderType_DayViewComponent)),
    import0.ɵdid(2482176,(null as any),0,import18.DayViewComponent,[
      import14.CalendarService,
      import0.ElementRef
    ]
    ,{
      dayviewAllDayEventTemplate: [
        0,
        'dayviewAllDayEventTemplate'
      ]
      ,
      dayviewNormalEventTemplate: [
        1,
        'dayviewNormalEventTemplate'
      ]
      ,
      dayviewAllDayEventSectionTemplate: [
        2,
        'dayviewAllDayEventSectionTemplate'
      ]
      ,
      dayviewNormalEventSectionTemplate: [
        3,
        'dayviewNormalEventSectionTemplate'
      ]
      ,
      formatHourColumn: [
        4,
        'formatHourColumn'
      ]
      ,
      formatDayTitle: [
        5,
        'formatDayTitle'
      ]
      ,
      allDayLabel: [
        6,
        'allDayLabel'
      ]
      ,
      hourParts: [
        7,
        'hourParts'
      ]
      ,
      eventSource: [
        8,
        'eventSource'
      ]
      ,
      markDisabled: [
        9,
        'markDisabled'
      ]
      ,
      locale: [
        10,
        'locale'
      ]
      ,
      dateFormatter: [
        11,
        'dateFormatter'
      ]
      ,
      dir: [
        12,
        'dir'
      ]
      ,
      scrollToHour: [
        13,
        'scrollToHour'
      ]
      ,
      preserveScrollPosition: [
        14,
        'preserveScrollPosition'
      ]
      ,
      lockSwipeToPrev: [
        15,
        'lockSwipeToPrev'
      ]
      ,
      lockSwipes: [
        16,
        'lockSwipes'
      ]
      ,
      startHour: [
        17,
        'startHour'
      ]
      ,
      endHour: [
        18,
        'endHour'
      ]
      ,
      spaceBetween: [
        19,
        'spaceBetween'
      ]
      ,
      hourSegments: [
        20,
        'hourSegments'
      ]

    }
    ,{
      onRangeChanged: 'onRangeChanged',
      onEventSelected: 'onEventSelected',
      onTimeSelected: 'onTimeSelected',
      onTitleChanged: 'onTitleChanged'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.dayviewAllDayEventTemplate || import0.ɵnov((<any>v.parent),6));
    const currVal_2:any = (co.dayviewNormalEventTemplate || import0.ɵnov((<any>v.parent),8));
    const currVal_3:any = (co.dayviewAllDayEventSectionTemplate || import0.ɵnov((<any>v.parent),12));
    const currVal_4:any = (co.dayviewNormalEventSectionTemplate || import0.ɵnov((<any>v.parent),14));
    const currVal_5:any = co.formatHourColumn;
    const currVal_6:any = co.formatDayTitle;
    const currVal_7:any = co.allDayLabel;
    const currVal_8:any = co.hourParts;
    const currVal_9:any = co.eventSource;
    const currVal_10:any = co.markDisabled;
    const currVal_11:any = co.locale;
    const currVal_12:any = co.dateFormatter;
    const currVal_13:any = co.dir;
    const currVal_14:any = co.scrollToHour;
    const currVal_15:any = co.preserveScrollPosition;
    const currVal_16:any = co.lockSwipeToPrev;
    const currVal_17:any = co.lockSwipes;
    const currVal_18:any = co.startHour;
    const currVal_19:any = co.endHour;
    const currVal_20:any = co.spaceBetween;
    const currVal_21:any = co.hourSegments;
    ck(v,1,1,[
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19,
      currVal_20,
      currVal_21
    ]
    );
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).class;
    ck(v,0,0,currVal_0);
  });
}
export function View_CalendarComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵpid(0,import7.DatePipe,[import0.LOCALE_ID]),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'monthviewDefaultDisplayEventTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_1)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'monthviewDefaultEventDetailTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_2)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'defaultAllDayEventTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_8)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'defaultNormalEventTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_9)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'defaultWeekViewAllDayEventSectionTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_10)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'defaultDayViewAllDayEventSectionTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_14)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'defaultNormalEventSectionTemplate',
        2
      ]
    ],(null as any),0,(null as any),View_CalendarComponent_17)),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import7.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_21)),
    import0.ɵdid(139264,(null as any),0,import7.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import7.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_22)),
    import0.ɵdid(139264,(null as any),0,import7.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import7.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CalendarComponent_23)),
    import0.ɵdid(139264,(null as any),0,import7.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import7.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import19.CalendarComponent = v.component;
    const currVal_1:any = co.calendarMode;
    ck(v,17,0,currVal_1);
    const currVal_2:any = 'month';
    ck(v,20,0,currVal_2);
    const currVal_3:any = 'week';
    ck(v,23,0,currVal_3);
    const currVal_4:any = 'day';
    ck(v,26,0,currVal_4);
  },(ck,v) => {
    var co:import19.CalendarComponent = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.calendarMode,'view-container');
    ck(v,16,0,currVal_0);
  });
}
function View_CalendarComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'calendar',([] as any[]),(null as any),(null as any),(null as any),View_CalendarComponent_0,RenderType_CalendarComponent)),
    import0.ɵprd(256,(null as any),import14.CalendarService,import14.CalendarService,([] as any[])),
    import0.ɵdid(122880,(null as any),0,import19.CalendarComponent,[
      import14.CalendarService,
      import0.LOCALE_ID
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const CalendarComponentNgFactory:import0.ComponentFactory<import19.CalendarComponent> = import0.ɵccf('calendar',import19.CalendarComponent,View_CalendarComponent_Host_0,{
  currentDate: 'currentDate',
  eventSource: 'eventSource',
  calendarMode: 'calendarMode',
  formatDay: 'formatDay',
  formatDayHeader: 'formatDayHeader',
  formatDayTitle: 'formatDayTitle',
  formatWeekTitle: 'formatWeekTitle',
  formatMonthTitle: 'formatMonthTitle',
  formatWeekViewDayHeader: 'formatWeekViewDayHeader',
  formatHourColumn: 'formatHourColumn',
  showEventDetail: 'showEventDetail',
  startingDayMonth: 'startingDayMonth',
  startingDayWeek: 'startingDayWeek',
  allDayLabel: 'allDayLabel',
  noEventsLabel: 'noEventsLabel',
  queryMode: 'queryMode',
  step: 'step',
  timeInterval: 'timeInterval',
  autoSelect: 'autoSelect',
  markDisabled: 'markDisabled',
  monthviewDisplayEventTemplate: 'monthviewDisplayEventTemplate',
  monthviewInactiveDisplayEventTemplate: 'monthviewInactiveDisplayEventTemplate',
  monthviewEventDetailTemplate: 'monthviewEventDetailTemplate',
  weekviewAllDayEventTemplate: 'weekviewAllDayEventTemplate',
  weekviewNormalEventTemplate: 'weekviewNormalEventTemplate',
  dayviewAllDayEventTemplate: 'dayviewAllDayEventTemplate',
  dayviewNormalEventTemplate: 'dayviewNormalEventTemplate',
  weekviewAllDayEventSectionTemplate: 'weekviewAllDayEventSectionTemplate',
  weekviewNormalEventSectionTemplate: 'weekviewNormalEventSectionTemplate',
  dayviewAllDayEventSectionTemplate: 'dayviewAllDayEventSectionTemplate',
  dayviewNormalEventSectionTemplate: 'dayviewNormalEventSectionTemplate',
  dateFormatter: 'dateFormatter',
  dir: 'dir',
  scrollToHour: 'scrollToHour',
  preserveScrollPosition: 'preserveScrollPosition',
  lockSwipeToPrev: 'lockSwipeToPrev',
  lockSwipes: 'lockSwipes',
  locale: 'locale',
  startHour: 'startHour',
  endHour: 'endHour',
  spaceBetween: 'spaceBetween'
}
,{
  onCurrentDateChanged: 'onCurrentDateChanged',
  onRangeChanged: 'onRangeChanged',
  onEventSelected: 'onEventSelected',
  onTimeSelected: 'onTimeSelected',
  onTitleChanged: 'onTitleChanged'
}
,([] as any[]));
